script.js

```diff
@@ -1,4 +1,341 @@
+var dealerTurn = "dealerTurn";
+var checkWinner = "checkWinner";
+
+var generateDeck = function () {
+  // create an array of 4 suits in the deck
+  var suits = ["Diamonds", "Clubs", "Hearts", "Spades"];
+
+  // while function to loop over the suits array
+  var suitIndex = 0;
+  while (suitIndex < suits.length) {
+    var currentSuit = suits[suitIndex];
+
+    // loop from 1 to 13 to create all the cards for the suits
+
+    var rankCounter = 1;
+    while (rankCounter <= 13) {
+      // by default card name is the same as rankcounter
+      var cardName = rankCounter;
+      var cardPoint = rankCounter;
+
+      // set the card value for ace, king, wueens
+      if (cardName == 1) {
+        cardName = "ace";
+      } else if (cardName == 11) {
+        cardName = "jack";
+        cardPoint = 10;
+      } else if (cardName == 12) {
+        cardName = "queen";
+        cardPoint = 10;
+      } else if (cardName == 13) {
+        cardName = "king";
+        cardPoint = 10;
+      }
+      // create a new card with the current name, suit and rank
+      var card = {
+        name: cardName,
+        suit: currentSuit,
+        rank: rankCounter,
+        point: cardPoint,
+      };
+      // add new card to the deck
+      cardDeck.push(card);
+      // increment rankCounter to iterate over the next rank
+      rankCounter += 1;
+    }
+    // increment suit index to iterate over the next suit
+    suitIndex += 1;
+  }
+  // return the completed card deck
+  return cardDeck;
+};
+
+// generate a random index from 0 (inclusive) to max (exclusive)
+var getRandomIndex = function (max) {
+  return Math.floor(Math.random() * max);
+};
+//   shuffle cards
+var shuffleCards = function (cardDeck) {
+  // loop over the card deck array
+  var currentIndex = 0;
+  while (currentIndex < cardDeck.length) {
+    // select a random index in the deck
+    var randomIndex = getRandomIndex(cardDeck.length);
+    var randomCard = cardDeck[randomIndex];
+    var currentCard = cardDeck[currentIndex];
+    // swap position of random card and current card in the deck
+    cardDeck[currentIndex] = randomCard;
+    cardDeck[randomIndex] = currentCard;
+    // increment currentIndex
+    currentIndex = currentIndex + 1;
+  }
+  // retutn the shuffled deck
+  shuffledDeck = cardDeck;
+  return shuffledDeck;
+};
+
+// deal 2 cards each to player and dealer from the shuffled deck
+var drawCard = function (cardDeck) {
+  var numOfCard = 2;
+  var cardCounter = 0;
+  while (cardCounter < numOfCard) {
+    playerHand.push(shuffledDeck.pop());
+    dealerHand.push(shuffledDeck.pop());
+    cardCounter += 1;
+  }
+
+  var dealedCards = `player Hand ${playerHand[0].name} of ${playerHand[0].suit} ${playerHand[1].name} of ${playerHand[1].suit} <br> dealer Hand  ${dealerHand[0].name} of ${dealerHand[0].suit}, ${dealerHand[1].name} of ${dealerHand[1].suit} `;
+  return dealedCards;
+};
+
+// compare player and dealer hand and return result.
+
+var checkResult = function () {
```

Comment:
There is alot of repetition in this function. You are checking if either the player or computer busts. Why not create a function called isBusted(hand) that returns a boolean if the hand is above 21?